### PR TITLE
[guilib] Add Container.TotalWatched and Container.TotalUnwatched 

### PR DESCRIPTION
**Pull request by BigNoid**
My original usecase for this was to be able to get the watched/unwatched count for the Home window widgets or playlist menu items, but since Home is not a media window, that doesn't work. Still these two infolabels can be useful in the library views.
For J****

@ronie @phil65

### CODE REVIEW

**Comment by BigNoid** [comment]
@ronie thx for review. I updated the pr and also removed setting it for music items, because unwatched makes no sense there and always is empty.
I'm seeing now that totalunwatched gets set to ```15.0-RC1``` in cases where there is no playcount. I'll update the pr if i find a fix for that.

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -1842,6 +1844,41 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
...
+      for (int i = 0; i<items.Size(); ++i)
+      {
+        CFileItemPtr item = items.Get(i);
+        if (item->HasVideoInfoTag() && (item->GetVideoInfoTag()->m_playCount == 0))
+          totalUnwatched += 1;
+      }
+      if (totalUnwatched > 0)
+        return StringUtils::Format("%i", totalUnwatched);
+    }
+  }
```
Comment (Montellese) [inline xbmc/GUIInfoManager.cpp]: You're missing a `break;` here which is causing the odd value if empty.

File: xbmc/GUIInfoManager.cpp [hunk 2]
```diff
@@ -1842,6 +1844,41 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
       }
     }
     break;
+  case CONTAINER_TOTALWATCHED:
+  {
```
Comment (Montellese) [inline xbmc/GUIInfoManager.cpp]: Please use the same indentation as above (same for `CONTAINER_TOTALUNWATCHED`).

**Comment by BigNoid** [comment]
@Montellese Updated, thx.
@ronie Indeed it should return zero for cases there are watched and no unwatched items or vice versa. I was following the totaltime logic by leaving it empty, so it only shows where applicable. It now returns zero for all directories, but its up to skinners to only show this where its applicable then.

File: xbmc/GUIInfoManager.cpp [hunk 3]
```diff
@@ -1842,6 +1844,40 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
...
+        for (int i = 0; i < items.Size(); ++i)
+        {
+          CFileItemPtr item = items.Get(i);
+          if (item->HasVideoInfoTag() && (item->GetVideoInfoTag()->m_playCount > 0))
+            playCount += 1;
+        }
+        return StringUtils::Format("%i", playCount);
+      }
+    }
+  break;
```
Comment (Montellese) [inline xbmc/GUIInfoManager.cpp]: Still bad indentation on both `break;` lines ;-)

File: xbmc/GUIInfoManager.cpp [hunk 4]
```diff
@@ -1842,6 +1844,40 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
...
+  case CONTAINER_TOTALUNWATCHED:
+    {
+      CGUIWindow *window = GetWindowWithCondition(contextWindow, WINDOW_CONDITION_IS_MEDIA_WINDOW);
+      if (window)
+      {
+        const CFileItemList& items = ((CGUIMediaWindow *)window)->CurrentDirectory();
+        int totalUnwatched = 0;
+        for (int i = 0; i<items.Size(); ++i)
```
Comment (Montellese) [inline xbmc/GUIInfoManager.cpp]: Spaces around the `<` please.

File: xbmc/GUIInfoManager.cpp [hunk 5]
```diff
@@ -1842,6 +1844,40 @@ std::string CGUIInfoManager::GetLabel(int info, int contextWindow, std::string *
...
+    {
+      CGUIWindow *window = GetWindowWithCondition(contextWindow, WINDOW_CONDITION_IS_MEDIA_WINDOW);
+      if (window)
+      {
+        const CFileItemList& items = ((CGUIMediaWindow *)window)->CurrentDirectory();
+        int playCount = 0;
+        for (int i = 0; i < items.Size(); ++i)
+        {
+          CFileItemPtr item = items.Get(i);
+          if (item->HasVideoInfoTag() && (item->GetVideoInfoTag()->m_playCount > 0))
```
Comment (Montellese) [inline xbmc/GUIInfoManager.cpp]: No need for the extra `(` and `)` around the second condition (same below).

**Comment by BigNoid** [comment]
@Montellese Updated, thx

**Comment by phil65** [comment]
Wouldnt it be nicer to "merge"  ```CONTAINER_TOTALWATCHED``` and ```CONTAINER_TOTALUNWATCHED``` instead of duplicating code?
(either with an if clause for the counting or by doing watched = items.size() - unwatched at the end)

**Comment by mkortstiege** [comment]
+1 for merging since it's just the operand that differs here.

**Comment by BigNoid** [comment]
That is much cleaner indeed, updated

**Comment by BigNoid** [comment]
@phil65 Had to do the if clause, as items.size()-playCount would also include the parent item (and any other item that has no playcount).

**Comment by phil65** [comment]
You can use the same counter variable for both calls, looks bit confusing atm.

**Comment by BigNoid** [comment]
Using one counter variable now and also merged with totaltime.
thx @phil65 for the pointer.

**Comment by phil65** [comment]
No problem.
@mkortstiege or @Montellese should decide though if merging with TOTALTIME is welcome.

**Comment by mkortstiege** [comment]
I see no issue with merging besides it's a bit harder to follow. @Montellese objections?

**Comment by mkortstiege** [comment]
Will shove this on in on monday if there are no further objections. jenkins build this please

**Comment by mkortstiege** [comment]
jenkins build this please

**Pull request merged**